lua/octo/workflow_runs.lua

```diff
@@ -0,0 +1,686 @@
+    M.refresh()
+  else
+    update_job_details(id)
+  end
+end
+
+---@param node WorkflowNode
+---@return string
+local function format_node(node)
+  local status = node.type == "step" and get_step_status(node.status, node.conclusion)
+    or node.type == "job" and get_job_status(node.status, node.conclusion)
+    or ""
+
+  local indent = string.rep(" ", node.indent)
+  local preIcon = node.type ~= "step_log" and (node.expanded == true and "> " or "> ") or ""
+  local formatted = string.format("%s%s%s %s", indent, preIcon, node.display, status)
+  return formatted
+end
+
+---@param node WorkflowNode
+---@param list LineDef[] | nil
+---@return LineDef[]
+local function tree_to_string(node, list)
+  list = list or {}
+  local formatted = format_node(node)
+  ---@type LineDef
+  local lineDef = {
+    value = formatted,
+    id = node.id,
+    type = node.type,
+    highlight = node.highlight,
+    step_log = nil,
+    expanded = node.expanded or false,
+    node_ref = node,
+  }
+
+  table.insert(list, lineDef)
+  if node.type ~= "step_log" then
+    table.insert(list, separator)
+  end
+
+  if node.expanded and next(node.children) then
+    for _, child in ipairs(node.children) do
+      tree_to_string(child, list)
+    end
+  end
+
+  return list
+end
+
+local function print_lines()
+  if not vim.api.nvim_buf_is_valid(M.buf) then
+    return
+  end
+  vim.api.nvim_buf_clear_namespace(M.buf, namespace, 0, -1)
+  vim.api.nvim_buf_set_option(M.buf, "modifiable", true)
+  local lines = get_workflow_header()
+
+  local stringified_tree = tree_to_string(M.tree, {})
+  for _, value in ipairs(stringified_tree) do
+    table.insert(lines, value)
+  end
+
+  local highlights = {}
+
+  local string_lines = {}
+
+  for index, line_def in ipairs(lines) do
+    table.insert(string_lines, line_def.value)
+    table.insert(highlights, { index = index - 1, highlight = line_def.highlight })
+  end
+
+  vim.api.nvim_buf_set_lines(M.buf, 0, -1, true, string_lines)
+  vim.api.nvim_buf_set_option(M.buf, "modifiable", false)
+
+  for _, vl in ipairs(highlights) do
+    if vl.highlight then
+      vim.api.nvim_buf_add_highlight(M.buf, namespace, vl.highlight, vl.index, 0, -1)
+    end
+  end
+
+  for binding, cb in pairs(tree_keymaps) do
+    vim.keymap.set("n", binding, function()
+      local current_line = vim.api.nvim_win_get_cursor(0)[1]
+      local line = lines[current_line]
+      if line.node_ref ~= nil then
+        cb(line.node_ref)
+      end
+    end, { silent = true, noremap = true, buffer = M.buf })
+  end
+  for binding, cb in pairs(keymaps) do
+    vim.keymap.set("n", binding, function()
+      cb(M)
+    end, { silent = true, noremap = true, buffer = M.buf })
+  end
+end
+
+M.refresh = function()
+  print_lines()
+end
+
+local workflow_limit = 100
+
+local function get_workflow_runs_sync(co)
+  local lines = {}
+  vim.fn.jobstart(
+    "gh run list --json conclusion,displayTitle,event,headBranch,name,number,status,updatedAt,databaseId -L "
```

Comment:
Possible to use gh.run / gh.run.list here as well? 
Using that ensures that the environment variables are correct